snscrape/modules/telegram.py

```diff
@@ -125,20 +219,45 @@ def _soup_to_items(self, soup, pageUrl, onlyUsername = False):
 					else:
 						_logger.warning(f'Could not process link preview image on {url}')
 				linkPreview = LinkPreview(**kwargs)
-			yield TelegramPost(url = url, date = date, content = content, outlinks = outlinks, linkPreview = linkPreview)
+				if kwargs['href'] in outlinks:
+					outlinks.remove(kwargs['href'])
```

Comment:
I'd prefer leaving the link preview href in `outlinks` as well, similar to how the Twitter scraper will have outlinks from link cards in `outlinks`.